pkg/prometheus/metrics.go

```diff
@@ -59,11 +80,16 @@ func init() {
 	legacyregistry.MustRegister(authFormCounterResult)
 	legacyregistry.MustRegister(authBasicCounter)
 	legacyregistry.MustRegister(authBasicCounterResult)
+	legacyregistry.MustRegister(x509MissingSANCounter)
 
 	for _, resultLabel := range []string{SuccessResult, FailResult, ErrorResult} {
 		authBasicCounterResult.WithLabelValues(resultLabel)
 		authFormCounterResult.WithLabelValues(resultLabel)
 	}
+
+	for _, provider := range []Provider{ProviderGitHub, ProviderGitLab, ProviderGoogle, ProviderOpenID, ProviderKeystone, ProviderBasicAuth} {
```

Comment:
mini-nit:
- `[...]Provider{ProviderGitHub, ProviderGitLab, ProviderGoogle, ProviderOpenID, ProviderKeystone, ProviderBasicAuth}`

Reply:
i like it, let's make it an array.